sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -449,6 +452,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
       sparkSession.sessionState.conf.sessionLocalTimeZone,
       sparkSession.sessionState.conf.columnNameOfCorruptRecord)
 
+    userSpecifiedSchema.foreach(checkJsonSchema)
```

Comment:
What about the case we read it from files?

Reply:
I'm not sure exactly what do you mean, I have insert the check in two points of `DataFrameReader`, in `def json(paths: String*): DataFrame` is where I undestand we read from files. I have also introduced a test. 
Do you mean something else?
Thanks!